### PR TITLE
Protocol 12.72 update

### PR DESCRIPTION
**Pull request by EPuncker**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- 12.72 support
- cid being dependent on GUID (this is necessary to save containers)
- adjust monster/npc generated ids (the `cid`)
- standardized creature id bounds (uses variable instead of hardcoded number now)
- improved code behind mounts
- updated summon status refresh packet
- updated talkactions message types (blue console message is deprecated)
- feature: packet sequencing
- feature: login with email (you can set `@admin` as your email btw or if you're running otclient, still use your account name)
- feature: right click to talk with npc
- feature: save opened containers
- feature: mount colors
- feature: session end packet
- feature: cid and GUID in onLook (gm/god only)
- new items
- new outfits
- new mounts
- new NPC bubbles
- new condition icons (enums only for now)
- new message types

### Credits
99% of the code on this PR is made by @Zbizu !!!

Special thanks to:
- OTBR community (especially people involved in "Global" and "Canary" projects and their OTClient fork) for providing the code and tools necessary to achieve that
- SaiyansKing for paving the way for QT clients
- Arch-Mina for providing version 12 tools to work with client
- EPuncker - providing spr/dat, help with otb, game content information, testing and help with login.php
- kokokoko - updating items.otb with new items and data
- omarcopires - review, testing and help with login.php
- TFS maintainers for ensuring code quality and helping with project direction

**Issues addressed:** 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Review by nekiro — APPROVED** [review]

File: src/protocolgame.cpp [hunk 1]
```diff
@@ -318,28 +318,43 @@ void ProtocolGame::logout(bool displayEffect, bool forced)
...
+	// Client type and OS used
 	OperatingSystem_t operatingSystem = static_cast<OperatingSystem_t>(msg.get<uint16_t>());
-	version = msg.get<uint16_t>();
 
-	msg.skipBytes(7); // U32 client version, U8 client type, U16 dat revision
+	version = msg.get<uint16_t>(); // U16 client version
+	msg.skipBytes(4); // U32 client version
 
+	// String client version
+	if (version >= 1240) {
```
Comment (omarcopires) [inline src/protocolgame.cpp:343]: as minor versions are not supported, do we still need this check?

File: src/protocolgame.cpp [hunk 2]
```diff
@@ -356,16 +372,31 @@ void ProtocolGame::onRecvFirstMessage(NetworkMessage& msg)
 		writeToOutputBuffer(opcodeMessage);
 	}
 
-	msg.skipBytes(1); // gamemaster flag
+	// Change packet verifying mode for QT clients
+	if (version >= 1111 && operatingSystem >= CLIENTOS_QT_LINUX && operatingSystem <= CLIENTOS_QT_MAC) {
```
Comment (omarcopires) [inline src/protocolgame.cpp:376]: as minor versions are not supported, do we still need this check?

Comment (nekiro) [reply src/protocolgame.cpp:376]: I would say no

Comment (Zbizu) [reply src/protocolgame.cpp:376]: Yes. This is needed to generate response about client version properly.

**Review by DSpeichert — CHANGES_REQUESTED** [review]
* Last commit ("fix mount login speed") should be squashed with the earlier one ("Fixed mount speed update")
* Typo in "Added senClientFeatures function"

**Comment by DSpeichert** [comment]
Linked #3771

**Review by DSpeichert — APPROVED** [review]

**Pull request merged**